### PR TITLE
Support TYPE flag in the SCAN command

### PR DESCRIPTION
**Pull request by jackyyyyyssss**
support  scan  additional type

### CODE REVIEW

File: src/commands/scan_base.h [hunk 1]
```diff
@@ -56,7 +56,12 @@ class CommandScanBase : public Commander {
           return {Status::RedisParseErr, "limit should be a positive integer"};
         }
       } else if (IsScan && parser.EatEqICase("type")) {
-        return {Status::RedisParseErr, "TYPE flag is currently not supported"};
+        std::string key = GET_OR_RET(parser.TakeStr());
+        if (RedisNamesMapType.count(key)) {
+          type_ = RedisNamesMapType.at(key);
+        } else {
+          return {Status::RedisExecErr, "Invalid type"};
+        }
```
Comment (PragmaTwice) [inline src/commands/scan_base.h]:
```suggestion
        if (auto iter = std::find(RedisTypeNames.begin(), RedisTypeNames.end(), type_str); iter != RedisTypeNames.end()) {
          type_ = static_cast<RedisType>(iter - RedisTypeNames.begin());
        } else {
          return {Status::RedisExecErr, "Invalid type"};
        }
```

File: src/commands/scan_base.h [hunk 2]
```diff
@@ -56,7 +56,12 @@ class CommandScanBase : public Commander {
           return {Status::RedisParseErr, "limit should be a positive integer"};
         }
       } else if (IsScan && parser.EatEqICase("type")) {
-        return {Status::RedisParseErr, "TYPE flag is currently not supported"};
+        std::string key = GET_OR_RET(parser.TakeStr());
```
Comment (PragmaTwice) [inline src/commands/scan_base.h]:
```suggestion
        std::string type_str = GET_OR_RET(parser.TakeStr());
```

what's `key`?

File: src/storage/redis_metadata.h [hunk 3]
```diff
@@ -94,6 +95,12 @@ enum RedisCommand {
 const std::vector<std::string> RedisTypeNames = {"none",   "string",    "hash",   "list",      "set",      "zset",
                                                  "bitmap", "sortedint", "stream", "MBbloom--", "ReJSON-RL"};
 
+const std::map<std::string, RedisType> RedisNamesMapType = {
+    {"none", kRedisNone},     {"string", kRedisString}, {"hash", kRedisHash},
+    {"list", kRedisList},     {"set", kRedisSet},       {"zset", kRedisZSet},
+    {"bitmap", kRedisBitmap}, {"stream", kRedisStream}, {"MBbloom--", kRedisBloomFilter},
+    {"ReJSON-RL", kRedisJson}};
+
```
Comment (PragmaTwice) [inline src/storage/redis_metadata.h]: Please remove this.

Comment (jackyyyyyssss) [reply src/storage/redis_metadata.h]: Thank you for your suggestion. I originally intended this design to reduce time complexity, but do you think it's not necessary to have too few array elements

Comment (PragmaTwice) [reply src/storage/redis_metadata.h]: Yeah, the maintenance cost will be far more than the performance benefit we get.

Comment (jackyyyyyssss) [reply src/storage/redis_metadata.h]: I understand. Thank you for your answer. I also missed an unverified type of test case. I will run the test case on the local branch first, and once it passes, please take a look on your end

**Review by PragmaTwice — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

**Review by jihuayu — APPROVED** [review]
Good work, thank you!

**Pull request merged**